lib/vdsm/storage/lvmdevices.py

```diff
@@ -52,6 +52,33 @@ def is_configured():
     return _enabled() and _devices_file_exists()
 
 
+def configure(vgs):
+    """
+     Configure lvm to use devices file and create initial system devices file.
+     If creation of initial devices file fails, disable it again if we enable
+     it.
+    """
+    devices_enabled = _enabled()
+    if not devices_enabled:
+        log.debug("lvm devices/use_devicesfile is not enabled, enabling it.")
+        _configure_devices_file(enable=True)
+    else:
+        log.debug("lvm devices/use_devicesfile is already enabled.")
+
+    # Always configure devices file. File maybe be empty or not up to date.
+    # On the other hand configuring correct devices file doesn't cause any
+    # harm.
+
+    try:
+        _create_system_devices(vgs)
+    except cmdutils.Error:
+        log.warning("Failed to create system devices file.")
+        if not devices_enabled:
+            log.debug("Disabling lvm devices/use_devicesfile.")
+            _configure_devices_file(enable=False)
```

Comment:
We know that devices file is enabled now, since we just enabled it, so we can just disable it. Even if another process disabled behind our back, disabling it when it is disabled in cheap and we don't need to complicate the code.

Reply:
Nice!